ln1_f4.py

```diff
@@ -8,86 +8,174 @@
-                           mem_cached, hdd_total, hdd_used, hdd_free]
-sysinfo_database_keys = ["cpu_total", "cpu_user", "cpu_sys", "cpu_idle", "mem_total", "mem_used", "mem_free",
-                         "mem_cached", "hdd_total", "hdd_used", "hdd_free"]
+    command_data, error_data = command_data.communicate()
+    return command_data.replace(',', '.').split()
+
+
+def get_cpu_info():  # return list format - cpu_user, cpu_sys, cpu_total, cpu_idle
+    cpu_console_data = get_sysresinfo('mpstat')
+    cpu_user = float(cpu_console_data[21])
+    cpu_sys = sum(float(x) for x in cpu_console_data[22:29])
+    cpu_total = cpu_user + cpu_sys
+    cpu_idle = float(cpu_console_data[30])
+    return [cpu_user, cpu_sys, cpu_total, cpu_idle]
+
+
+def get_mem_info():  # return list format - mem_total, mem_used, mem_free, mem_cached
+    mem_console_data = get_sysresinfo('free -m')
+    mem_total, mem_used = int(mem_console_data[7]), int(mem_console_data[8])
+    mem_free, mem_cached = int(mem_console_data[9]), int(mem_console_data[12])
+    return [mem_total, mem_used, mem_free, mem_cached]
 
-###############################################################################################################
 
-if os.path.isfile("/home/lax/PycharmProjects/learn1/database_cron") == False:  # Adding new record into database
-    with open("/home/lax/PycharmProjects/learn1/database_cron", "w") as database_file:
-        sysinfo_database = {"0": {
-            "email_send_hour": 24
-        }}
-        database_file.write(json.dumps(sysinfo_database))
-
-with open("/home/lax/PycharmProjects/learn1/database_cron", "r") as database_file:
-    data = database_file.read().strip()
-sysinfo_database = json.loads(data)
-sysdb_totalrecords = len(sysinfo_database)
-sysinfo_database[str(sysdb_totalrecords)] = {
-    "record_time": systime_customformat,
-    "cpu_total": cpu_total,
-    "cpu_user": cpu_user,
-    "cpu_sys": cpu_sys,
-    "cpu_idle": cpu_idle,
-    "mem_total": mem_total,
-    "mem_used": mem_used,
-    "mem_free": mem_free,
-    "mem_cached": mem_cached,
-    "hdd_total": hdd_total,
-    "hdd_used": hdd_used,
-    "hdd_free": hdd_free
-}
-with open("/home/lax/PycharmProjects/learn1/database_cron", "w") as database_file:
-    database_file.write(json.dumps(sysinfo_database))
+def get_hdd_info():  # return list format - hdd_total, hdd_used, hdd_free
+    hdd_console_data = get_sysresinfo('df -m --total')
+    hdd_total, hdd_used = int(hdd_console_data[50]), int(hdd_console_data[51])
+    hdd_free = int(hdd_console_data[52])
+    return [hdd_total, hdd_used, hdd_free]
 
 ###############################################################################################################
-dynamic_html_table = """
+
+
+class SysinfoDatabase(object):
+    def __init__(self, db_file_name):
+        self.db_file_name = db_file_name
+        if not os.path.isfile(self.db_file_name):
```

Comment:
try to avoid double file open (lines 60-71). You could change that lines with that:
```
mode = 'r' if os.path.isfile(self.db_file_name) else 'w'
file = open(filename, mode)
self.sys_info = file.read().strip()
if self.sys_info:
    self.sys_info = json.loads(self.sys_info)
else:
    self.sys_info = {}
```